src/abstractdataframe/abstractdataframe.jl

```diff
@@ -3291,3 +3291,62 @@ function insertcols!(df::AbstractDataFrame; after::Bool=false,
     _drop_all_nonnote_metadata!(parent(df))
     return df
 end
+
+"""
+    Iterators.partition(df::AbstractDataFrame, n)
+
+Iterate over `df` data frame `n` rows at a time, returning each block
+as a `SubDataFrame`.
+
+# Examples
+
+```jldoctest
+julia> collect(Iterators.partition(DataFrame(x=1:5), 2))
+3-element Vector{SubDataFrame{DataFrame, DataFrames.Index, UnitRange{Int64}}}:
+ 2×1 SubDataFrame
+ Row │ x
+     │ Int64
+─────┼───────
+   1 │     1
+   2 │     2
+ 2×1 SubDataFrame
+ Row │ x
+     │ Int64
+─────┼───────
+   1 │     3
+   2 │     4
+ 1×1 SubDataFrame
+ Row │ x
+     │ Int64
+─────┼───────
+   1 │     5
+```
+"""
+function Iterators.partition(df::AbstractDataFrame, n::Integer)
+    n < 1 && throw(ArgumentError("cannot create partitions of length $n"))
+    return Iterators.PartitionIterator(df, Int(n))
+end
+
+# use autodetection of eltype
+Base.IteratorEltype(::Type{<:Iterators.PartitionIterator{<:AbstractDataFrame}}) =
+    Base.EltypeUnknown()
+
+# we do not need to be overly specific here as we rely on autodetection of eltype
+Base.eltype(::Type{<:Iterators.PartitionIterator{<:AbstractDataFrame}}) =
+    AbstractDataFrame
+
+IteratorSize(::Type{<:Iterators.PartitionIterator{<:AbstractDataFrame}}) =
+    Base.HasLength()
```

Comment:
Isn't this redundant with the default behavior of `PartitionIterator`?

Reply:
I want it for completeness in one place (e.g. in case that this changes in `Iterators` module in the future)